src/Compilers/CSharp/Test/Semantic/Semantics/ColorColorTests.cs

```diff
@@ -2063,5 +2063,80 @@ void M()
                 Diagnostic(ErrorCode.ERR_VariableUsedBeforeDeclaration, "E").WithArguments("E").WithLocation(11, 17)
                 );
         }
+
+        [WorkItem(19458, "https://github.com/dotnet/roslyn/issues/19458")]
+        [Fact]
+        public void VersionUnificationConstColorColorEnum()
+        {
+            string sourceRefLib = @"
+public enum Color { Red }
+";
+
+            var refLib = CreateCompilation(
+                sourceRefLib,
+                assemblyName: "RefLib",
+                references: new[] { TestReferences.NetFx.v2_0_50727.mscorlib });
+
+            refLib.VerifyEmitDiagnostics();
+
+            string sourceMain = @"
+class M
+{
+    void F()
+    {
+        const Color Color = Color.Red;
+        _ = Color; // to avoid unused local warning
+    }
+}
+";
+
+            var main = CreateCompilation(
+                sourceMain,
+                assemblyName: "Main",
+                references: new MetadataReference[]
+                {
+                    new CSharpCompilationReference(refLib),
+                    TestReferences.NetFx.v4_0_30319.mscorlib
+                });
+
+            var unifyReferenceWarning =
+                // warning CS1701: Assuming assembly reference 'mscorlib, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089' used by 'RefLib' matches identity 'mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089' of 'mscorlib', you may need to supply runtime policy
+                Diagnostic(ErrorCode.WRN_UnifyReferenceMajMin).WithArguments(
+                    "mscorlib, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089",
+                    "RefLib",
+                    "mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089",
+                    "mscorlib");
+
+            main.VerifyEmitDiagnostics(unifyReferenceWarning, unifyReferenceWarning, unifyReferenceWarning, unifyReferenceWarning, unifyReferenceWarning);
+        }
+
+        [WorkItem(19458, "https://github.com/dotnet/roslyn/issues/19458")]
+        [Fact]
+        public void ObsoleteConstColorColorEnum()
+        {
+            string source = @"
+enum Color
+{
+    [System.Obsolete] Red
+}
+
+class M
+{
+    void F()
+    {
+        const Color Color = Color.Red;
+        _ = Color; // to avoid unused local warning
+    }
+}
+";
+
+            var compilation = CreateStandardCompilation(source, assemblyName: "Main");
+
+            DiagnosticDescription obsoleteWarning =
+                // warning CS0612: 'Color.Red' is obsolete
+                Diagnostic(ErrorCode.WRN_DeprecatedSymbol, "Color.Red").WithArguments("Color.Red").WithLocation(11, 29);
+
+            compilation.VerifyEmitDiagnostics(obsoleteWarning, obsoleteWarning);
```

Comment:
> How did the behavior of those tests change with the fix?

They stopped producing the "Failed to emit module" error diagnostic. [Here is full output of the two tests without the fix.](https://gist.githubusercontent.com/svick/b93124bce29944435e442dbd10cad0d6/raw/9a44f7417396a0e903c415ed0b183645142bae95/ConstColorEnumTests.txt)

(I believe you could have seen this in the CI results for the first commit, but those are now 404.)

> From the description of the bug, it feels like we should also emit the compilation (not just check the emit diagnostics).

I don't understand. As far as I can tell, `VerifyEmitDiagnostics` internally already calls `Compilation.Emit`.

Are you saying I should also call `Emit` directly? What should I assert about the output of that? What would be the point of that?